File: app/src/main/java/se/curtrune/lucy/classes/Notification.java
package se.curtrune.lucy.classes;

import static se.curtrune.lucy.util.Logger.log;

import androidx.annotation.NonNull;

import com.google.gson.Gson;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.Locale;

import se.curtrune.lucy.util.Converter;

public class Notification implements Serializable {
    public enum Type{
        PENDING, ALARM, NOTIFICATION
    }
    private Type type;
    private long date;
    private int time;
    private String title;
    private String content;
    public  Notification(){
        log("Notification()");
        type = Type.NOTIFICATION;
    }
    public String getContent(){
        return this.content;
    }
    public LocalDate getDate(){
        return LocalDate.ofEpochDay(date);
    }
    public String getTitle(){
        return this.title;
    }
    public LocalTime getTime(){
        return LocalTime.ofSecondOfDay(time);
    }
    public Type getType(){
        return this.type;
    }
    public void setContent(String content){
        this.content = content;
    }
    public void setDate(LocalDate date){
        this.date = date.toEpochDay();
    }
    public void setDate(String string) {
        date = LocalDate.parse(string).toEpochDay();
    }


    public void setTitle(String title){
        this.title = title;
    }
    public void setTime(LocalTime time){
        this.time = time.toSecondOfDay();
    }
    public void setTime(String string) {
        time = LocalTime.parse(string).toSecondOfDay();
    }
    public void setType(Type type){
        this.type = type;
    }
    public String toJson(){
        return new Gson().toJson(this);
    }

    @NonNull
    @Override
    public String toString() {
        return String.format(Locale.getDefault(), "%s %s %s", type.toString(), getDate().toString(), Converter.format(getTime()));
    }
}

File: app/src/main/java/se/curtrune/lucy/features/notifications/Notification.kt
package se.curtrune.lucy.features.notifications

import se.curtrune.lucy.util.Converter
import se.curtrune.lucy.util.Logger.Companion.log
import java.io.Serializable
import java.time.LocalDate
import java.time.LocalTime
import java.util.Locale

@kotlinx.serialization.Serializable
class Notification {
    enum class Type {
        PENDING, ALARM, NOTIFICATION
    }

    @JvmField
    var type: Type
    private var date: Long = 0
    private var time = 0
    @JvmField
    var title: String? = null
    @JvmField
    var content: String? = null

    init {
        if (VERBOSE) log("Notification()")
        type = Type.NOTIFICATION
    }

    fun getDate(): LocalDate {
        return LocalDate.ofEpochDay(date)
    }

    fun getTime(): LocalTime {
        return LocalTime.ofSecondOfDay(time.toLong())
    }

    fun setDate(date: LocalDate) {
        this.date = date.toEpochDay()
    }

    fun setDate(string: String?) {
        date = LocalDate.parse(string).toEpochDay()
    }


    fun setTime(time: LocalTime) {
        this.time = time.toSecondOfDay()
    }

    fun setTime(string: String?) {
        time = LocalTime.parse(string).toSecondOfDay()
    }


    override fun toString(): String {
        return String.format(
            Locale.getDefault(),
            "%s %s %s",
            type.toString(),
            getDate().toString(),
            Converter.format(getTime())
        )
    }

    companion object {
        var VERBOSE: Boolean = false
    }
}
